webapp/login/views.py

```diff
@@ -72,6 +78,17 @@ def after_login(resp):
 
 @login.route('/logout')
 def logout():
+    no_redirect = flask.request.args.get('no_redirect')
+
     if authentication.is_authenticated(flask.session):
         authentication.empty_session(flask.session)
-    return flask.redirect('/')
+
+    if no_redirect == 'true':
+        return flask.redirect('/')
+    else:
+        return flask.redirect(BSI_URL + '/auth/logout')
+
+
+@login.route('/login/bsi')
```

Comment:
I think this shouldn't be in the same PR as it would cause issues when deploying. See comment on: https://github.com/canonical-websites/build.snapcraft.io/pull/1129/files

Reply:
Agreed, moving to separate PR